AmazonPublisherServicesAdapter/src/main/java/com/chartboost/mediation/amazonpublisherservicesadapter/AmazonPublisherServicesAdapter.kt

```diff
@@ -864,3 +775,133 @@ class AmazonPublisherServicesAdapter : PartnerAdapter {
         }
     }
 }
+
+private class AdListener(
+    private val continuationRef: WeakReference<CancellableContinuation<Result<PartnerAd>>>,
+    private val request: PartnerAdLoadRequest,
+    private val listenerRef: WeakReference<PartnerAdListener?>,
+) : DTBAdInterstitialListener {
+    private lateinit var fullscreenAd: DTBAdInterstitial
```

Comment:
this holds onto the activity, though. which should be fine, maybe?
what if you load an amazon placement, rotate the screen to kill the current activity, and try to show that ad?
also, do you need this to be lateinit? Why not just make it `DTBAdInterstitial?`

Reply:
kk changed to the nullable type. I don't have fills anymore so can't test the rotation thing.